cli/cli/kurtosis_config/resolved_config/kurtosis_cluster_config.go

```diff
@@ -20,9 +23,10 @@ var dockerBackendApiContainerModeArgs *backend_creator.APIContainerModeArgs = ni
 type kurtosisBackendSupplier func(ctx context.Context) (backend_interface.KurtosisBackend, error)
 
 type KurtosisClusterConfig struct {
```

Comment:
can we slap a comment on top on what 

kurtosisBackendSupplier, kurtosisRemoteBackendConfigSupplier & engineBackendConfigSupplier are and used for

Reply:
The whole thing is super messy right now. I think it mostly comes from the fact that the CLI is backend aware. Most of this complexity comes from the fact that the CLI has its own backend, but it also provides the config for the engine to instantiate its backend... -.-
So, basically here 
- kurtosisBackendSupplier is the supplier for the CLI backend
- engineBackendConfigSupplier is the supplier for the config the engine will use to instantiate its own backend
And now we're adding:
- kurtosisRemoteBackendConfigSupplier is the supplier of the config for the remote backend the engine should connect to, if any

I think making the CLI non backend aware will be a big win here. This might be possible now that we have the portal, although it requires everyone to run the portal locally, which will not happen tomorrow. So, basically I'm good not spending too many cycles refactoring this to something better.

But I generally agree it's confusing, took me some time to fully understand those, so I'll stick a good comment here to make sure future us get ramped up on this quickly